cpp/include/cuml/distance/distance_type.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+namespace ML {
+namespace Distance {
+
+/** enum to tell how to compute euclidean distance */
+enum DistanceType : unsigned short {
+  /** evaluate as dist_ij = sum(x_ik^2) + sum(y_ij)^2 - 2*sum(x_ik * y_jk) */
+  EucExpandedL2 = 0,
+  /** same as above, but inside the epilogue, perform square root operation */
+  EucExpandedL2Sqrt = 1,
+  /** cosine distance */
+  EucExpandedCosine = 2,
```

Comment:
There's cosine distance and then there's cosine similarity. Do you know if we need to support both?

Reply:
Sklearn only supports cosine distance since all reported values in `pairwise_distance` are the difference between the inputs, not the similarity. I dont think its necessary to support both since its easy to convert between the two. I think its more important to be consistent across the different metrics that are available to the method.

Reply:
Also, if we ever wanted to do something like expose our “sortColsInRows” prim to allow the independent ordering of rows by distance (Eg for kNN-like operation), it would be a very quick elementwise operation to invert the distance back into a similarity.

Reply:
Just to be clear, are we OK with keeping the implementation in this PR? i.e. Using the distance, not the similarity.

Reply:
Resolving based on conversation with Corey